### DISCUSSION TITLE
如何通过环境变量限定上下文的长度？

### DISCUSSION THREAD

**Discussion by HowardTangOvO** [lobehub/lobehub#2783] [Self Hosting] [answered]
### 📦 Environment

- [ ] Official
- [ ] Official Preview
- [X] Vercel / Zeabur / Sealos
- [ ] Docker
- [ ] Other

### 📌 Version

Latest Version

### 💻 Operating System

- [ ] Windows
- [ ] macOS
- [ ] Ubuntu
- [ ] Other Linux
- [ ] iOS
- [ ] Android
- [ ] Other

### 🌐 Browser

- [ ] Chrome
- [ ] Edge
- [ ] Safari
- [ ] Firefox
- [ ] Other

### 🐛 Bug Description

<img width="702" alt="image" src="https://github.com/lobehub/lobe-chat/assets/22932158/8a880feb-df48-4208-bd0e-2b680e750189">
在Vercel 当中配置了`DEFAULT_AGENT_CONFIG` 为`params.max_tokens=32768;` 实际并未生效
Tokens还是爆掉了

[a comment by an automated account is not shown]

**Comment by HowardTangOvO** [comment]
看到源码当中, AgentConfig如果并没有enableMaxTokens则不可以开启max_tokens?
```js
agentConfig.params.max_tokens = chatConfig.enableMaxTokens
      ? agentConfig.params.max_tokens
      : undefined;
 ```
 看起来是需要在客户端开启`enableMaxTokens` 即`开启单次回复限制` 才会生效，
 能否在Environment Variables 也加上enableMaxTokens的开关呢？

[a comment by an automated account is not shown]

**Comment by arvinxx** [comment]
试试 `params.max_tokens=32768;chatConfig.enableMaxTokens=true`

**Comment by HowardTangOvO** [comment]
@arvinxx 
<img width="927" alt="image" src="https://github.com/lobehub/lobe-chat/assets/22932158/0c7a9b21-8256-496b-a493-39ecffc8d378">

<img width="613" alt="image" src="https://github.com/lobehub/lobe-chat/assets/22932158/6412923d-33f2-44ec-9ff0-5b02269c6d10">

配置上了并且重新构建了，还是没有生效@@

**Comment by arvinxx** [comment]
`max_tokens` 不是上下文的长度…是生成文本的最大长度

[a comment by an automated account is not shown]

**Comment by HowardTangOvO** [comment, marked as answer]
那如何可以限定上下文的长度？
现在出现的问题就是上下文长度过长导致tokens消耗得很快
根据你前面的写法的话可以配置`chatConfig.enableHistoryCount=true;chatConfig.historyCount=5` 来配置对吗？

**Comment by arvinxx** [comment]
> 根据你前面的写法的话可以配置chatConfig.enableHistoryCount=true;chatConfig.historyCount=5 来配置对吗？

对
